argo/common/run_completion.go

```diff
@@ -20,18 +25,52 @@ var RunCompletionStatuses = struct {
 }
 
 type RunCompletionEvent struct {
-	Status       RunCompletionStatus `json:"status"`
-	PipelineName NamespacedName      `json:"pipelineName"`
+	Status       RunCompletionStatus `json:"status" validate:"required"`
+	PipelineName NamespacedName      `json:"pipelineName" validate:"pipelineName"`
 	// Optionally render structs until https://github.com/golang/go/issues/11939 is addressed
-	RunConfigurationName  *NamespacedName `json:"runConfigurationName,omitempty"`
-	RunName               *NamespacedName `json:"runName,omitempty"`
-	RunId                 string          `json:"runId"`
+	RunConfigurationName  *NamespacedName `json:"runConfigurationName"`
+	RunName               *NamespacedName `json:"runName"`
+	RunId                 string          `json:"runId" validate:"required"`
 	ServingModelArtifacts []Artifact      `json:"servingModelArtifacts"`
 	Artifacts             []Artifact      `json:"artifacts,omitempty"`
-	Provider              string          `json:"provider"`
+	Provider              string          `json:"provider" validate:"required"`
 }
 
 func (sre RunCompletionEvent) String() string {
 	return fmt.Sprintf("{Status:%s PipelineName:%+v RunConfigurationName:%+v RunName:%+v RunId:%s ServingModelArtifacts:%+v Artifacts:%+v Provider:%s}",
 		sre.Status, sre.PipelineName, sre.RunConfigurationName, sre.RunName, sre.RunId, sre.ServingModelArtifacts, sre.Artifacts, sre.Provider)
 }
+
+func validateNamespacedName(nn *NamespacedName, key string) error {
+	if nn == nil {
+		return fmt.Errorf("key: %s is nil", key)
+	} else if nn.Name == "" {
+		return fmt.Errorf("key: %s, Name filed is missing", key)
```

Comment:
Looks like there is still a typo??

Reply:
No idea how it has been missed even if we commit suggestions.

Reply:
328d1937212e144ecb6c10f47f5f81b38edfd5c3